### PR TITLE
feat(vite-node): clear console when rendering error get fixed

### LINKED ISSUE #4787: Clear console on HMR
Currently, the terminal only outputs the warnings and errors but has no feedback on HMR events. Making it stays on the previous error even when the app is back to working normally.

### PR DESCRIPTION
**Pull request by antfu**
### 🔗 Linked issue

fix #4787

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Atinux** [comment]
How do you test it locally?

**Comment by antfu** [comment]
You can add a `throw new Error` in `<script setup>` to trigger SSR error and then remove it.

**Comment by pi0** [comment]
Nice idea to only clear console when there are errors but i think we can do it with main consola instance wrapping console in nuxi to catch all sources of warning and error.

As for this PR, can we show a message when we detect vite-node assets are changed in next render? Currently there is no watcher and message like client-side indicating HMR happened or legacy vite builder. Do we still have a watcher with vite-node ?

**Comment by Atinux** [comment]
I may do something wrong but I don't see the clear (also got the server stuck at one point):

Uploading Screenshot 2022-08-02 at 17.12.17.mp4…

File: packages/vite/src/runtime/vite-node.mjs [hunk 1]
```diff
@@ -28,5 +30,12 @@ export default async (ssrContext) => {
       runner.moduleCache.delete(key)
     }
   }
+  if (ssrContext.error) {
+    previousError = ssrContext.error
+  } else if (previousError) {
+    previousError = null
+    consola.clear()
```
Comment (pi0) [inline packages/vite/src/runtime/vite-node.mjs:37]: This doesn't works to clear screen. `consola.clear` was legacy name for `consola. removeReporter`. 

Also we probably shouldn't clear screen from runtime.

**Comment by pi0** [comment]
Thanks to #6343, we can now show accurate message on HMR. Moving log enhancenment to #6351. And lets track clear on rebuild experience in https://github.com/nuxt/framework/issues/4787. I think we could handle it on nuxi level.

**Pull request closed without merging**